src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.ApplicationModel.Configuration/ConfigurationClient.cs

```diff
@@ -63,29 +63,26 @@ public async Task<Response<ConfigurationSetting>> AddAsync(ConfigurationSetting
 
             Uri uri = BuildUriForKvRoute(setting);
 
-            using (HttpMessage message = _pipeline.CreateMessage(cancellation)) {
-                ReadOnlyMemory<byte> content = Serialize(setting);
-
-                message.SetRequestLine(HttpVerb.Put, uri);
+            var request = _pipeline.CreateRequest();
+            ReadOnlyMemory<byte> content = Serialize(setting);
 
-                message.AddHeader("Host", uri.Host);
-                message.AddHeader(IfNoneMatch, "*");
-                message.AddHeader(MediaTypeKeyValueApplicationHeader);
-                message.AddHeader(HttpHeader.Common.JsonContentType);
-                message.AddHeader(HttpHeader.Common.CreateContentLength(content.Length));
-                AddClientRequestID(message);
-                AddAuthenticationHeaders(message, uri, HttpVerb.Put, content, _secret, _credential);
+            request.SetRequestLine(HttpVerb.Put, uri);
 
-                message.SetContent(HttpMessageContent.Create(content));
+            request.AddHeader("Host", uri.Host);
+            request.AddHeader(IfNoneMatch, "*");
+            request.AddHeader(MediaTypeKeyValueApplicationHeader);
+            request.AddHeader(HttpHeader.Common.JsonContentType);
+            request.AddHeader(HttpHeader.Common.CreateContentLength(content.Length));
+            AddClientRequestID(request);
+            AddAuthenticationHeaders(request, uri, HttpVerb.Put, content, _secret, _credential);
 
-                await _pipeline.SendMessageAsync(message).ConfigureAwait(false);
+            request.SetContent(HttpMessageContent.Create(content));
 
-                var response = message.Response;
-                if (response.Status == 200 || response.Status == 201) {
-                    return await CreateResponse(response, cancellation);
-                }
-                else throw new RequestFailedException(response);
+            var response = await _pipeline.SendMessageAsync(request, cancellation).ConfigureAwait(false);;
```

Comment:
With `IDisposable` now on the `Response`, you may want to consider wrapping in a `using` block.   Going to stick with a single comment here, but each creation of the response below seems to follow the same pattern.

Reply:
The API consumer is responsible for disposing Response